transports/sslsocket.go

```diff
@@ -1,27 +1,139 @@
 package transports
 
 import (
+	"compress/zlib"
 	"crypto/tls"
 	"fmt"
+	"io"
 	"net"
+	"time"
 )
 
 type sslsocket struct {
 	net.Conn
 }
 
+type zlibConn struct {
```

Comment:
It seems the only difference between `zlibConn` and the existing Zlib transport is lazy initialization of the write part. And now I'm not sure that it is really necessary because you are doing here exactly what happens in raw socket - create a raw connection with `net.Dial` and then apply compression with `newZlibconn`